### PR TITLE
Add slot for ProgressHeader and ProgressTitle

### LINKED ISSUE #36463: Extensibility: Free trial welcome texts
P2: pdibGW-1iQ-p2

Image

The store’s welcome title and description is customized for free trial. We should use SlotFills for each of the text component.

Title is located [here](https://href.li/?https://github.com/woocommerce/woocommerce/blob/c3d62f113addbbf820ed5e8103a0a052092efc00/plugins/woocommerce-admin/client/task-lists/progress-title/progress-title.tsx#L65). 

Description is located [here](https://href.li/?https://github.com/woocommerce/woocommerce/blob/c3d62f113addbbf820ed5e8103a0a052092efc00/plugins/woocommerce-admin/client/task-lists/progress-header/progress-header.tsx#L55).

### Acceptance Criteria

1. An extension developer should be able to replace the title components
2. Extensibility point names should contain `experimental` keyword
2. Ideally, include sufficient documentation
1. Ideally, adhere to extensibility principles Pca54o-vO-p2 and pca54o-2ag-p2

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #36463 .

Added slot for [ProgressTitle](https://github.com/woocommerce/woocommerce/blob/c3d62f113addbbf820ed5e8103a0a052092efc00/plugins/woocommerce-admin/client/task-lists/progress-title/progress-title.tsx) and [Progressheader](https://github.com/woocommerce/woocommerce/blob/c3d62f113addbbf820ed5e8103a0a052092efc00/plugins/woocommerce-admin/client/task-lists/progress-header/progress-header.tsx) components.

### How to test the changes in this Pull Request:

1. Checkout this branch. 
2. [demo/homescreen-welcome-text-slot](https://github.com/woocommerce/woocommerce/tree/demo/homescreen-welcome-text-slot) branch to include examples. This branch adds `example-fills.tsx` in `progress-header` and `progress-title` directories.
3. Navigate to `WooCommerce -> Home`. You should see the following progress title and header.

Screen Shot 2023-01-17 at 11 57 35 AM

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by chihsuan — DISMISSED** [review]
Nice work! LGTM and tested well. 🚢

File: plugins/woocommerce-admin/client/task-lists/progress-header/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+import { Slot, Fill } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { createOrderedChildren, sortFillsByOrder } from '../../utils';
+
+export const WC_TASKLIST_PROGRESS_HEADER_SLOT_NAME =
+	'woocommerce_tasklist_progress_header_item';
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/task-lists/progress-header/utils.tsx]: Just learned that we should add `Experimental` somewhere. 😄 

https://github.com/woocommerce/woocommerce/pull/36487#discussion_r1073234410

**Review by rjchow — DISMISSED** [review]
nice 😀

File: plugins/woocommerce-admin/client/task-lists/progress-header/utils.tsx (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/task-lists/progress-header/utils.tsx]: @chihsuan good point! Updated in https://github.com/woocommerce/woocommerce/pull/36482/commits/d2950f87a7be03318a6211e58548673870ceaf0f

**Review by rjchow — APPROVED** [review]

**Review by chihsuan — APPROVED** [review]

**Pull request merged**